common/lib/capa/capa/responsetypes.py

```diff
@@ -1099,22 +1228,91 @@ def setup_response(self):
+        credit_type = problem_xml[0].get('partial_credit', default=False)
+
+        try:
+            credit_type = str(credit_type).lower().strip()
+        except ValueError:
+            _ = self.capa_system.i18n.ugettext
+            # Translators: 'partial_credit' is an attribute name and should not be translated.
+            msg = _("partial_credit value can only be set to 'points' or be removed.")
+            raise LoncapaProblemError(msg)
+        
+
         for aid in amap:
-            if aid in student_answers and student_answers[aid] == amap[aid]:
-                cmap.set(aid, 'correct')
-            else:
-                cmap.set(aid, 'incorrect')
+            for word in amap[aid]:
+                if aid in student_answers and student_answers[aid] == word:
+                    cmap.set(aid, 'correct')
+                    break
+                else:
+                    cmap.set(aid, 'incorrect')
+                    
+            # For partial credit:
+            if credit_type == 'points':
+                partial_map = self.get_partial()
+                points_map = self.get_partial_points(partial_map)
+
+                if not cmap.is_correct(aid) and partial_map[aid] is not None: 
+                    for index, word in enumerate(partial_map[aid]):
+                        if aid in student_answers and student_answers[aid] == word:
+                            cmap.set(aid, 'partially-correct')
+                            cmap.set_property(aid, 'npoints', points_map[aid][index])
+                            break
+                        else:
+                            cmap.set(aid, 'incorrect')
+                        
             answer_variable = self.get_student_answer_variable_name(student_answers, aid)
             if answer_variable:
                 cmap.set_property(aid, 'answervariable', answer_variable)
+                
         return cmap
 
     def get_answers(self):
         amap = dict([(af.get('id'), contextualize_text(af.get(
             'correct'), self.context)) for af in self.answer_fields])
+        # Split to allow multiple correct answers.
+        for aid in amap:
+            amap[aid] = amap[aid].split(',')
+            for index, word in enumerate(amap[aid]):
+                amap[aid][index] = word.strip()
         # log.debug('%s: expected answers=%s' % (unicode(self),amap))
         return amap
+    
+    def get_partial(self):
+        pmap = dict([(af.get('id'), contextualize_text(af.get(
+            'partial'), self.context)) for af in self.answer_fields])
+        # Split to allow multiple partially correct answers.
+        for aid in pmap:
+            if pmap[aid] is not None:
+                pmap[aid] = pmap[aid].split(',')
+                for index, word in enumerate(pmap[aid]):
+                    pmap[aid][index] = word.strip()
+        # log.debug('%s: partially correct answers=%s' % (unicode(self),amap))
```

Comment:
I tried to copy from existing code as much as possible when I was doing this (which is why there are comments about translation that I never would have thought about). That line is copied from the get_answers function above and adapted to the new get_partial function.